modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/introspection/validation/MetadataComponentModelValidatorTestCase.java

```diff
@@ -310,4 +319,23 @@ private ParameterModel getMockKeyPartParam(Object defaultValue, int order) {
     when(param.getModelProperty(MetadataKeyPartModelProperty.class)).thenReturn(of(new MetadataKeyPartModelProperty(order)));
     return param;
   }
+
+  public static class MockResolver implements TypeKeysResolver, OutputTypeResolver {
+
+    @Override
+    public MetadataType getOutputType(MetadataContext context, Object key)
+        throws MetadataResolvingException, ConnectionException {
+      return null;
```

Comment:
maybe something for a follow up ticket, but I think that any resolver should be forbidden of returning null.